src/common/CMakeLists.txt

```diff
@@ -23,5 +23,8 @@ file(GLOB_RECURSE SOURCES
 
 set(OBJ_LIB ${LIB_NAME}_common)
 add_library(${OBJ_LIB} OBJECT ${SOURCES})
+set_target_properties(${OBJ_LIB} PROPERTIES COMPILE_FLAGS "${DNNL_ARCH_OPT_FLAGS}")
```

Comment:
Can you please explain why this change is needed?

Reply:
For Open Image Denoise we're directly using the compile parameters from MKL-DNN, but we can't always use ISAs that may not be available on all platforms, e.g. SSE4.1, so we need to compile a portion of the library without these extra ISAs. Having the possibility to apply these ISA options to only certain source files enables us to detect ISA support in the library and report an error instead of crashing due to executing unsupported instructions. This change probably would not have direct benefits for MKL-DNN right now, but it would make adding this behavior straightforward in the future. Compiling the entire MKL-DNN library for a specific ISA is not flexible enough both performance and backward compatibility wise.

Reply:
Could we handle this by setting `CMAKE_CXX_FLAGS` in `src/CMakeLists.txt`? The issue with the current solution is that we have to duplicate `set_target_properties` for every sub-directory of `src`: `src/common`, `src/cpu`, `src/ocl` and any other directory we add in future.

On the other hand, if the goal is to limit `DNNL_ARCH_OPT_FLAGS` to the `src` directory we could just move `append(CMAKE_CXX_FLAGS "${DNNL_ARCH_OPT_FLAGS}")` from `cmake/platform.cmake` to `src/CMakeLists.txt`.

Reply:
Yes, moving `append(CMAKE_CXX_FLAGS "${DNNL_ARCH_OPT_FLAGS}")` to `src/CMakeLists.txt` should work too.

Reply:
Closing the discussion as the changes [are not relevant anymore](https://github.com/intel/mkl-dnn/pull/572#issuecomment-544695471)